pkg/cli/cli.go

```diff
@@ -193,6 +212,59 @@ func WithCompletion(c *cli) error {
 	return nil
 }
 
+//set default descriptions for root command
+func (c *cli) setRootCommandDefaults() error {
```

Comment:
Can you just do this checks in `func (c cli) defaultCommand() *cobra.Command`? I don't see the need to do it that early and I feel like you loose track of those fields by setting them at the start and using them at the end.